src/shared/hooks/useAccessDataForTable.tsx

```diff
@@ -360,6 +386,81 @@ const accessData = async (
   return { ...result, headerProperties, tableResource };
 };
 
+const resourceToLocalStorageResource = (resource: Resource): TDataSource => {
+  let localStorageRow: TDataSource;
+  try {
+    const resourceName = resource.name ?? resource['@id'] ?? resource._self;
+    localStorageRow = {
+      key: resource._self,
+      _self: resource._self,
+      id: resource['@id'],
+      name: resourceName,
+      project: resource._project,
+      description: resource.description ?? '',
+      createdAt: resource._createdAt,
+      updatedAt: resource._updatedAt,
+      source: 'studios',
+      type: isArray(resource['@type'])
+        ? resource['@type']
+        : [resource['@type'] ?? ''],
+
+      distribution: {
+        contentSize: isArray(resource.distribution)
+          ? sumBy(
+              resource.distribution,
+              distItem => distItem.contentSize?.value ?? 0
+            )
+          : resource.distribution?.contentSize ?? 0,
+        encodingFormat: isArray(resource.distribution)
+          ? resource.distribution.find(distItem => distItem.encodingFormat)
+              ?.encodingFormat ?? ''
+          : resource.distribution?.encodingFormat ?? '',
+        label: isArray(resource.distribution)
+          ? resource.distribution.map(distItem => {
+              const itemLabel =
+                distItem.label ??
+                `${resourceName}.${fileExtensionFromResourceEncoding(
+                  distItem.encodingFormat,
+                  'txt'
+                )}`;
+              return itemLabel;
+            })
+          : resource.distribution?.label ??
+            `${resourceName}.${fileExtensionFromResourceEncoding(
+              resource.distribution?.encodingFormat,
+              'txt'
```

Comment:
why a default file extension as `txt`, this may damage the file if it's not a txt, 
I prefer to let it as it is (no extension) and the user can decide which app can open the file (The user know better the data he/she has)

Reply:
agreed. I don't add a default extension now.